modules/runner-binaries-syncer/main.tf

```
@@ -4,44 +4,55 @@ locals {
 
 resource "aws_s3_bucket" "action_dist" {
   bucket        = var.distribution_bucket_name
-  acl           = "private"
   force_destroy = true
   tags          = var.tags
+}
 
-  # Max 1 block - server_side_encryption_configuration
-  dynamic "server_side_encryption_configuration" {
-    for_each = length(keys(var.server_side_encryption_configuration)) == 0 ? [] : [var.server_side_encryption_configuration]
-
-    content {
-
-      dynamic "rule" {
-        for_each = length(keys(lookup(server_side_encryption_configuration.value, "rule", {}))) == 0 ? [] : [lookup(server_side_encryption_configuration.value, "rule", {})]
+resource "aws_s3_bucket_acl" "action_dist_acl" {
+  bucket = aws_s3_bucket.action_dist.id
+  acl    = "private"
+}
 
-        content {
-          bucket_key_enabled = lookup(rule.value, "bucket_key_enabled", null)
+resource "aws_s3_bucket_lifecycle_configuration" "bucket-config" {
+  bucket = aws_s3_bucket.action_dist.id
 
-          dynamic "apply_server_side_encryption_by_default" {
-            for_each = length(keys(lookup(rule.value, "apply_server_side_encryption_by_default", {}))) == 0 ? [] : [
-            lookup(rule.value, "apply_server_side_encryption_by_default", {})]
+  rule {
+    id     = "lifecycle_config"
+    status = "Enabled"
 
-            content {
-              sse_algorithm     = apply_server_side_encryption_by_default.value.sse_algorithm
-              kms_master_key_id = lookup(apply_server_side_encryption_by_default.value, "kms_master_key_id", null)
-            }
-          }
-        }
-      }
+    abort_incomplete_multipart_upload {
+      days_after_initiation = 7
     }
-  }
-
-  lifecycle_rule {
-    enabled                                = true
-    abort_incomplete_multipart_upload_days = 7
 
     transition {
       days          = 35
       storage_class = "INTELLIGENT_TIERING"
     }
+
+
+  }
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "action_dist" {
+  bucket = aws_s3_bucket.action_dist.id
+
+
+  dynamic "rule" {
+    for_each = length(keys(lookup(var.server_side_encryption_configuration, "rule", {}))) == 0 ? [] : [lookup(var.server_side_encryption_configuration, "rule", {})]
```

```suggestion
  count = length(keys(lookup(var.server_side_encryption_configuration, "rule", {}))) == 0 ? 0 : 1

  dynamic "rule" {
    for_each = length(keys(lookup(var.server_side_encryption_configuration, "rule", {}))) == 0 ? [] : [lookup(var.server_side_encryption_configuration, "rule", {})]
```

  dynamic "rule" {
    for_each = [lookup(var.server_side_encryption_configuration, "rule", {})]

it is not allowed to define a `aws_s3_bucket_server_side_encryption_configuration` resource without any rule:
`aws_s3_bucket_server_side_encryption_configuration`

the resource must be omitted completely when no rule ist defined

---

@shashidhar087

---

Good catch, thans @julada I will commit the suggested change